integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,186 @@
+        self.clickup_url = clickup_url
+        self.clickup_token = clickup_token
+        self.api_url = f"{self.clickup_url}/api/v2"
+        self.client = http_async_client
+        self.client.timeout = Timeout(30)
+        self.team_key = None
+
+    async def initialize_team_key(self):
+        self.team_key = await self.get_team_key()
+
+    @property
+    async def api_headers(self) -> dict[str, Any]:
+        return {
+            "Authorization": self.clickup_token,
+            "Content-Type": "application/json",
+        }
+
+    async def get_team_key(self) -> List[str]:
+        """Get the team key for the Clickup account."""
+        url = f"{self.api_url}/team"
+        response = await self._send_api_request(url)
+        teams = response.get("teams", [])
+        team_ids = []
+        for team in teams:
+            team_ids.append(team.get("id"))
+        return team_ids
+
+    async def _send_api_request(
+        self,
+        url: str,
+        params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+        method: str = "GET",
+    ) -> Any:
+        """Send a request to the Clickup API."""
+        logger.debug(f"Sending request {method} to endpoint {url}")
+        response = await self.client.request(
+            url=url,
+            method=method,
+            headers=await self.api_headers,
+            params=params,
+            json=json_data,
+        )
+        response.raise_for_status()
+        return response.json()
+
+    async def get_single_clickup_team(self, team_id: str) -> dict[str, Any]:
+        """Get a single Clickup team by team_id."""
+        url = f"{self.api_url}/team/{team_id}"
+        response = await self._send_api_request(url)
+        team = response.get("team")
+        return team
+
+    async def get_clickup_teams(self) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all Clickup teams."""
+        url = f"{self.api_url}/team"
+        response = await self._send_api_request(url)
+        teams = response.get("teams", [])
+        yield teams
+
+    async def _get_spaces_in_team(self, team_id: str) -> AsyncGenerator[str, None]:
+        """Get all spaces in a workspace."""
+        url = f"{self.api_url}/team/{team_id}/space"
+        params = {"archived": "false"}
+        response = await self._send_api_request(url, params)
+        spaces = response.get("spaces")
+        for space in spaces:
+            yield space.get("id")
+
+    async def _get_folders_in_space(self, team_id: str) -> AsyncGenerator[str, None]:
+        """Get all folders in a space."""
+        async for space in self._get_spaces_in_team(team_id):
+            url = f"{self.api_url}/space/{space}/folder"
+            params = {"archived": "false"}
+            response = await self._send_api_request(url, params)
+            folders = response.get("folders")
+            for folder in folders:
+                yield folder.get("id")
```

Comment:
Just return the whole folders response instead of this

Reply:
Done